lib/actions/jlinkTargetActions.js

```diff
@@ -467,18 +500,23 @@ const recoverOneCore = (serialNumberWithCore, core) => async (
     }
     dispatch(targetActions.erasingStartAction());
     const { family } = getState().app.target.deviceInfo;
-    if (family === 'nRF53') {
-        const prgFunc = () =>
-            new Promise((resolve, reject) =>
+    if (family === 'nRF53' || family === 'nRF52') {
+        let recoverFile;
+        recoverFile = family === 'nRF52' ? recover52hex : recoverFile;
+        recoverFile =
```

Comment:
You already know that the `family` is either 'nRF52' or 'nRF53' from the if-statement above, so this can be simplified to ```js
const recoverFile = family === 'nRF52' ? recover52hex : recover53hex[core.coreNumber];
```

Reply:
I have thought about writing like this, but there are also several other stuff in the if block, do you have a suggestion for that?